sdks/python/apache_beam/transforms/enrichment_handlers/vertex_ai_feature_store.py

```diff
@@ -0,0 +1,289 @@
+from apache_beam.transforms.enrichment import EnrichmentSourceHandler
+from apache_beam.transforms.enrichment_handlers.utils import ExceptionLevel
+
+__all__ = [
+    'VertexAIFeatureStoreEnrichmentHandler',
+    'VertexAIFeatureStoreLegacyEnrichmentHandler'
+]
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def _not_found_err_message(
+    feature_store_name: str, feature_view_name: str, entity_id: str) -> str:
+  """returns a string formatted with given parameters"""
+  return (
+      "make sure the Feature Store: %s with Feature View "
+      "%s has entity_id: %s" %
+      (feature_store_name, feature_view_name, entity_id))
+
+
+class VertexAIFeatureStoreEnrichmentHandler(EnrichmentSourceHandler[beam.Row,
+                                                                    beam.Row]):
+  """Enrichment handler to interact with Vertex AI Feature Store.
+
+  Use this handler with :class:`apache_beam.transforms.enrichment.Enrichment`
+  transform when the Vertex AI Feature Store is set up for
+  Bigtable Online serving.
+
+  With the Bigtable Online serving approach, the client fetches all the
+  available features for an entity-id. To filter the features to enrich, use
+  the `join_fn` param in :class:`apache_beam.transforms.enrichment.Enrichment`.
+
+  **NOTE:** The default severity to report exceptions is logging a warning. For
+    this handler, Vertex AI client returns the same exception
+    `Requested entity was not found` even though the feature store doesn't
+    exist. So make sure the feature store instance exists or set
+    `exception_level` as `ExceptionLevel.RAISE`.
+  """
+  def __init__(
+      self,
+      project: str,
+      location: str,
+      api_endpoint: str,
+      feature_store_name: str,
+      feature_view_name: str,
+      row_key: str,
+      *,
+      exception_level: ExceptionLevel = ExceptionLevel.WARN,
+      **kwargs,
+  ):
+    """Initializes an instance of `VertexAIFeatureStoreEnrichmentHandler`.
+
+    Args:
+      project (str): The GCP project-id for the Vertex AI Feature Store.
+      location (str): The region for the Vertex AI Feature Store.
+      api_endpoint (str): The API endpoint for the Vertex AI Feature Store.
+      feature_store_name (str): The name of the Vertex AI Feature Store.
+      feature_view_name (str): The name of the feature view within the
+        Feature Store.
+      row_key (str): The row key field name containing the unique id
+        for the feature values.
+      exception_level: a `enum.Enum` value from
+        `apache_beam.transforms.enrichment_handlers.utils.ExceptionLevel`
+        to set the level when an empty row is returned from the BigTable query.
+        Defaults to `ExceptionLevel.WARN`.
+      kwargs: Optional keyword arguments to configure the
+        `aiplatform.gapic.FeatureOnlineStoreServiceClient`. When using
+        `kwargs`, the `api_endpoint` param will be overridden with this config
+        if multiple values are found for `client_options`.
```

Comment:
Instead of this, can we ad api_endpoint to the `client_options` (if its non-empty) and throw if it already exists/is different?

Reply:
We can do this in `__init__`

Reply:
Same comment applies below